### PR TITLE
48 add gmail service into the DB

### LINKED ISSUE #48: Add gmail actions/reactions, post them into the Database
**Short description:**
Find and add some gmail action/reactions

**Describe the solution you'd like**
Add the gmail service into the DB following the prisma schema,
find possible action/reactions which could be implemented

### PR DESCRIPTION
**Pull request by GuyomT**
# Purpose

The purpose of this pull request is to add the action search email from the service gmail
and the way to create the service gmail in the database.

Fixes #48 

## Added features

* This service has been added to the application. It includes:
  * A action which is used to search for an email following the api query format (an example in on discord in the gmail channel)
  * A function to create the service with prisma in the db

### CODE REVIEW

File: server/services/gmail/actions/read_email.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const gmail = require('../gmail_init').getGmailClient();
+
+async function readEmail(emailId) {
```
Comment (LucasTesnier) [inline server/services/gmail/actions/read_email.js]: Missing documentation

File: server/services/gmail/actions/read_email.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  try {
+    const email = await gmail.users.messages.get({userId: 'me', id: emailId});
+    console.log(email.data.snippet);
+  } catch (err) {
+    console.log(`An error occurred: ${err}`);
+  }
+}
+
+async function searchEmails(query) {
```
Comment (LucasTesnier) [inline server/services/gmail/actions/read_email.js]: Missing documentation

File: server/services/gmail/actions/read_email.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  try {
+    const response = await gmail.users.messages.list({userId: 'me', q: query});
+    const messages = response.data.messages;
+    if (messages.length) {
+      // iterate through the messages and get the id
```
Comment (LucasTesnier) [inline server/services/gmail/actions/read_email.js]: Useless documentation

File: server/services/gmail/actions/read_email.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      for (let i = 0; i < messages.length; i++) {
+        const email =
+            await gmail.users.messages.get({userId: 'me', id: messages[i].id});
+        console.log(email.data.id)
```
Comment (LucasTesnier) [inline server/services/gmail/actions/read_email.js]: Please remove console.log when merging to main

File: server/services/gmail/actions/read_email.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        console.log(email.data.snippet);
+      }
+    } else {
+      console.log('No email found.');
```
Comment (LucasTesnier) [inline server/services/gmail/actions/read_email.js]: Use some return keyword and not console.log

File: server/services/gmail/gmail_init.js [hunk 6]
```diff
@@ -1,7 +1,31 @@
 FROM_EMAIL = 'aequallsquared@gmail.com'
 
+const database = require('../../database_init')
 const {google} = require('googleapis');
 
+function createGmailService() {
```
Comment (LucasTesnier) [inline server/services/gmail/gmail_init.js]: Missing documentation

File: server/services/gmail/gmail_init.js [hunk 7]
```diff
@@ -1,7 +1,31 @@
...
+  const gmailService = {
+    name: 'Gmail',
+    description: 'Gmail service',
```
Comment (LucasTesnier) [inline server/services/gmail/gmail_init.js]: It will be updated in part 2 ?

File: server/services/gmail/gmail_init.js [hunk 8]
```diff
@@ -1,7 +1,31 @@
...
+    isEnable: false,
+    createdAt: new Date(),
+    Actions: [{
+      name: 'Read email',
+      description: 'Read an email',
```
Comment (LucasTesnier) [inline server/services/gmail/gmail_init.js]: Same ?

File: server/services/gmail/gmail_init.js [hunk 9]
```diff
@@ -1,7 +1,31 @@
...
+      isEnable: false,
+      createdAt: new Date(),
+    }],
+    Reactions: [{
+      name: 'Send email',
+      description: 'Send an email',
```
Comment (LucasTesnier) [inline server/services/gmail/gmail_init.js]: Same ?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
What's the utility of searchEmail function is she doesn't return anything ?

Please create some documentation

**Review by TomDUVAL-MAHE — DISMISSED** [review]
NTR, but you have conflicts !

**Review by LucasTesnier — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**